src/qcodes/dataset/data_set_cache.py

```diff
@@ -91,6 +92,30 @@ def data(self) -> ParameterData:
 
         return self._data
 
+    @staticmethod
+    def _empty_data_dict(
+        interdeps: InterDependencies_,
+    ) -> dict[str, dict[str, npt.NDArray]]:
+        """
+        Create an dictionary with empty numpy arrays as values
+        matching the expected output of ``DataSet``'s ``get_parameter_data`` /
```

Comment:
this is cool but "sorted" is used in those functions now, so the order is not the same between them and this empty dict data function, right? or am i not reading the code carefully?

Reply:
There is sorting at 2 levels. The most important one is the inner most one where the top_level parameter comes first and is followed by the dependencies in added order. QCoDeS it self was relying on this. I have removed this partially, but I think we should still maintain this for backwards compatibility. 

The other sorting is with the outer most keys that are simply sorted alphabetically. I don't think we rely on that other than perhaps some tests. I still think it makes sense to maintain that. Unfortunately there is no std lib ordered set so if we make top_level_params a tuple again if we wanted to keep the sorting there. That is probably the best option